src/commands/init.js

```diff
@@ -322,79 +321,35 @@ const addToXcodeBuild = async (cwd: string) => {
+  const haulSignature = 'added by Haul';
+
+  /* Make sure we check both project iOS and iOS-TV, that's the magic behind "2" constant */
+
+  const countOccurrences = search => {
+    return project.split(search).length - 1;
+  };
 
   // Are we already integrated?
-  if (project.includes(haulScriptKey)) {
+  if (countOccurrences(haulSignature) === 2) {
     progress.info('Haul is already part of your build scripts');
     return;
   }
 
-  /**
-   * Define Haul integration script in the PBXShellScriptBuildPhase section.
-   *
-   * This is done by prepending our predefined script phase to the list
-   * of all phases.
-   */
-  project = project.replace(
-    '/* Begin PBXShellScriptBuildPhase section */',
-    dedent`
-      /* Begin PBXShellScriptBuildPhase section */
-      ${haulScriptKey} /* Integrate Haul with React Native */ = {
-        isa = PBXShellScriptBuildPhase;
-        buildActionMask = 2147483647;
-        name = "Integrate Haul with React Native";
-        files = (
-        );
-        inputPaths = (
-        );
-        outputPaths = (
-        );
-        runOnlyForDeploymentPostprocessing = 0;
-        shellPath = /bin/sh;
-        shellScript = "bash ../node_modules/haul/src/utils/haul-integrate.sh";
-      };`
-  );
+  const originalTask = 'shellScript = "export NODE_BINARY=node';
 
-  /**
-   * Add Haul integration to build phases that already contain `react-native-xcode.sh`
-   *
-   * We are typically trying to match the following:
-   *
-   * ```
-   *   buildPhases = (
-   *     13B07F871A680F5B00A75B9A \/* Sources *\/,
-   *     13B07F8C1A680F5B00A75B9A \/* Frameworks *\/,
-   *     13B07F8E1A680F5B00A75B9A \/* Resources *\/,
-   *     00DD1BFF1BD5951E006B06BC \/* Bundle React Native code and images *\/,
-   *   );
-   * ```
-   *
-   * and prepend our build phase to the beginning.
-   */
-  let sectionsCount = 0;
-  project = project.replace(/buildPhases = \(\n(?:.*,\n)*\s*\);/g, match => {
-    if (!match.includes('React Native')) return match;
-    sectionsCount++;
-    return match.replace(
-      'buildPhases = (',
-      dedent`
-        buildPhases = (
-          ${haulScriptKey} /* Integrate Haul with React Native */,
-        `
-    );
-  });
-
-  if (sectionsCount > 0) {
-    fs.writeFileSync(path.join(entry, 'project.pbxproj'), project);
-    progress.succeed('Added haul to your Xcode build scripts');
-  } else {
-    progress.fail(
-      `Failed to add Haul to your Xcode build scripts. See: ${chalk.grey(
-        'https://github.com/callstack-io/haul/blob/master/docs/Configuring%20Your%20Project.md#integrating-with-xcode'
-      )} for manual instructions`
+  if (countOccurrences(originalTask) !== 2) {
+    progress.warn(
+      `Couldn't edit Xcode project. Haven't recognized 'Bundle React Native code and images' build phase.`
     );
+    return;
   }
+
+  const haulTask = `shellScript = "# ${haulSignature}\nexport CLI_PATH=node_modules/haul/bin/cli.js\nexport NODE_BINARY=node`;
```

Comment:
can replace \n with visual newlines, as we're in template literal

Reply:
let's leave it as one-liner I don't want to mess with Regex 👍